ibc/store/proofs_ics23.go

```diff
@@ -0,0 +1,159 @@
+package store
+
+import (
+	"crypto/sha256"
+
+	ics23 "github.com/cosmos/ics23/go"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/smt"
+)
+
+type position int
+
+const (
+	left  position = iota // 0
+	right                 // 1
+)
+
+var (
+	// Custom SMT spec as the store does not hash values
```

Comment:
I keep bringing this up because I'm still personally not 100% sure myself (but trust your judgment), but you're confident that we should keep hashing the other values even though we're not hashing these ones?

Lmk if this is better for an offline dicussion.

Reply:
We dont hash any other values. The other side nodes are already hashed. When we insert into the tree there are 3 steps:
1. hash the key - this gives us the path (we use the `PathHasher`)
  - The `PathHasher` can be customised with the option `WithPathHasher()` when creating the tree, this cannot be `nil`
2. hash the value - this gives us the valueHash (we use the `ValueHasher`)
  - The `ValueHasher` can be customised with the option `WithValueHasher()` if the value hasher is `nil` we do a `noop` function when hashing the value leaving us with the raw bytes
3. We then hash both of these using the `TreeHasher` this is required when we create the tree - this gives us the nodes digest
  - The `TreeHasher` is used as the `PathHasher` and `ValueHasher` if they are not provided
  - The `TreeHasher` cannot be nil

From this we can see that we dont want to prehash the value as we need the raw bytes but the node is still encoded and hashed to give the digest whether the value is or not. The side nodes are digests, they are already hashed. What we do with them is create a new inner node like this:
```
innerNode = []byte{[]byte{1}, leftChild.digest, rightChild.digest}
```
And then we hash that to give us the next digest.

Reply:
The hashing and nonhashing makes sense to me in the context of IBC and SMT in independence.

It's not a blocker for this PR, but my question/concern is moreso for the larger context of the blockchain to make sure that I'm personally not missing/misunderstanding something. For the rest of the persistence module, question is: `Should we hash the values for the other trees or not?`

@dylanlott do you have thoughts/opinion here?

Reply:
So in my ICS-24 PR I made the decision to not hash any values in all of the trees (except root tree as its created differently) this is just to make things easier during creation. But open to ideas here

Reply:
Interesting. Would like to discuss this more (as a team). 

Moving the thread to https://github.com/pokt-network/pocket/pull/847#discussion_r1245898305